python/ray/data/_internal/plan.py

```diff
@@ -90,6 +90,8 @@ def __init__(
         self._run_by_consumer = run_by_consumer
         self._dataset_name = None
 
+        self._has_executed = False
```

Comment:
IMO `_has_executed` makes it sound like the dataset has always finished executing (e.g. "The dataset has executed"), but looking at the code, it looks like it really means "has started execution." Is there another name we could use to indicate that the dataset execution may still be in progress?

Reply:
Good call. Renamed to `_has_started_execution`